openedx/core/djangoapps/course_apps/api.py

```diff
@@ -42,4 +56,9 @@ def set_course_app_enabled(course_key: CourseKey, app_id: str, enabled: bool, us
     """
     course_app = CourseAppsPluginManager.get_plugin(app_id)
     enabled = course_app.set_enabled(course_key, user=user, enabled=enabled)
+    CourseAppStatus.objects.update_or_create(
```

Comment:
@xitij2000 Do you think this call can still create objects in the data model? I am looking at the usage of `update_or_create` rather than just updating the instance? Can you explain it a little?
thanks.

Reply:
Yes, it can still create object, since it's possible (though unlikely) that a update will happen before the app is initialised. 

Since this is an open API, a custom plugin or integration using the REST of python API could trigger this code before a model exists.

Reply:
got it thanks.